### PR TITLE
WIP - Run Jenkins example using service account

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
@bparees PTAL

File: examples/jenkins/jenkins-config.json [hunk 1]
```diff
@@ -4,6 +4,13 @@
   "metadata": {},
   "items": [
     {
+      "kind": "ServiceAccount",
+      "apiVersion": "v1beta3",
```
Comment (bparees) [inline examples/jenkins/jenkins-config.json]: no v1 for this?

Comment (liggitt) [inline examples/jenkins/jenkins-config.json]: Rest of the file is v1beta3, thought we'd update the whole thing at once

Comment (bparees) [inline examples/jenkins/jenkins-config.json]: fair enough.

**Comment by bparees** [comment]
lgtm.  i'm working on pushing an updated jenkins image.  I guess we need to setup a job for that one too, or build it as part of origin.

**Comment by bparees** [comment]
oh i guess that one is on an automated build so your dummy PR should have triggered it.  not sure why it didn't..i'll try to kick one off.

**Comment by bparees** [comment]
@liggitt or @ncdc i'm assuming the curl commands at the end of this jenkins job (the one that check the registry and update tags and use the v1 docker api) probably don't work anymore either, right?  (due to needing an access token)

**Comment by bparees** [comment]
I wonder if the right way to do this now is to actually just update the imagestream directly (change the imagestreamtag or mapping) rather than poking the registry.

**Comment by liggitt** [comment]
Ha, hadn't gotten that far because of the API version.

**Comment by ncdc** [comment]
@bparees so it's trying to get the image id for the tag `test` and then it's trying to create a new tag `prod` using that id?

**Comment by liggitt** [comment]
Yeah, retag the test build as prod after it passes tests

**Comment by ncdc** [comment]
Instead of querying the registry for the latest image id for the tag, I would query OpenShift for the image used by the deployment. It should hopefully return a digest (`...@sha256:...`). If it does, great. If not, then I guess you could possibly query the registry, but you'd need to use credentials.

You have some options for promoting `test` to `prod`:

1. Use `oc tag`. This will set a spec tag and update status tags. The only problem with this is that it becomes a tracking tag, so anytime `test` is updated, `prod` will automatically by updated too (@smarterclayton FYI)
1. Use `docker login`, `docker tag`, `docker push`.
1. Try to post an ImageStreamMapping, but you might get permission denied depending on the policy settings.

**Comment by liggitt** [comment]
is there a way to tag a specific image?

**Comment by liggitt** [comment]
heh, I can rtfm I guess:
```
  // Tag a specific image:
  $ oc tag openshift/ruby@sha256:6b646fa6bf5e5e4c7fa41056c27910e679c03ebe7f93e361e6515a9da7e258cc yourproject/ruby:tip
```

**Pull request closed without merging**